tests/test_meta/test_estimatortransformer.py

```diff
@@ -85,3 +85,24 @@ def test_shape_multitarget(random_xy_dataset_multitarget):
         ]
     )
     assert pipeline.fit(X, y).transform(X).shape == (m, n)
+
+
+def test_kwargs(random_xy_dataset_clf):
+    """ Test if kwargs are properly passed to an underlying estimator. """
+    X, y = random_xy_dataset_clf
+    pipeline = Pipeline(
+        [
+            (
+                "model",
+                EstimatorTransformer(Ridge(random_state=0))
+            )
+        ]
+    )
+    # If the additional parameter sample_weight is passed transform_no_kwargs and transform_kwargs should not be equal.
+    transform_no_kwargs = pipeline.fit(X, y).transform(X)
+    transform_kwargs = pipeline.fit(X, y, model__sample_weight=0.1).transform(X)
+    assert not np.array_equal(transform_no_kwargs, transform_kwargs)
```

Comment:
I'm a bit surprised that the predictions are different when we apply a uniform weight to all the examples. Aren't we just dividing the loss function by a constant here?

Reply:
> This is advice unrelated to the PR, but in general, it's advisable to work on a branch that isn't main. Especially when you're potentially working on multiple features.
> 
> ![CleanShot 2022-09-14 at 15 46 16](https://user-images.githubusercontent.com/1019791/190171435-5e5facee-e62e-48b6-82d9-4e06aa168f89.png)

👍 That makes sense!

Thanks for all this feedback, really appreciate it!

Reply:
> This is advice unrelated to the PR, but in general, it's advisable to work on a branch that isn't main. Especially when you're potentially working on multiple features.
> 
> ![CleanShot 2022-09-14 at 15 46 16](https://user-images.githubusercontent.com/1019791/190171435-5e5facee-e62e-48b6-82d9-4e06aa168f89.png)

That makes sense, will create a new branch next time when working on a features. Thanks for all this feedback, really appreciate it!

Reply:
I suppose so, it's more that I'm worried about what the test is testing when it's assuming an unintuitive result to be true.

Reply:
[The data is rescaled](https://github.com/scikit-learn/scikit-learn/blob/36958fb240fbe435673a9e3c52e769f01f36bec0/sklearn/linear_model/_base.py#L326) and it seems to me here is where a constant sample_weight < 1 can change model training, namely learning less from each sample. I adjusted the test to use `LinearRegression` so its easier to analyse this. Rescaling data is being called [here](https://github.com/scikit-learn/scikit-learn/blob/36958fb240fbe435673a9e3c52e769f01f36bec0/sklearn/linear_model/_base.py#L702).

I also adjusted the test to use an array of sample weights (i.e. all samples weighted equally except the 1st sample has a weight of 2).

Reply:
Rather than checking whether the results are the same or different, I'd rather test more explicitly whether the right parameters are called. 

I think the best thing to do here would be to make the estimator passed to the EstimatorTransformers a `unittest.Mock` object with a `fit` method. Then we can use `assert_called_with` to verify the sample_weights are indeed passed. That way we make sure we test explicitly the parameter passing, rather than any other reason the results might be different (randomness for example)